uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -146,8 +146,14 @@ define([
      * @param  {LeaveRequestInstance} leaveRequest
      */
     function deleteLeaveRequest (leaveRequest) {
-      removeLeaveRequestFromIndexedList(leaveRequest);
-      updateLeaveRequestDaysContactData(leaveRequest);
+      var indexedLeaveRequest = findIndexedLeaveRequest(leaveRequest);
```

Comment:
what's the difference between a `leaveRequest` and an `indexedLeaveRequest` ? Why isn't this handled at the `removeLeaveRequestFromIndexedList` or `updateLeaveRequestDaysContactData` level?

Reply:
Very good question. This is because `updateLeaveRequestDaysContactData`, imagine you update a leave request `5 May - 10 May` to `4 May - 9 May`. It needs to flush days `5 - 10` and then add for `4 - 9`. We delete indexed request and not the one that was passed in order to flush the `10th May` as well. Then we add `4 May - 9 May`.